elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metric/BaseMetric.java

```diff
@@ -0,0 +1,142 @@
+
+import com.yahoo.elide.datastores.aggregation.Column;
+import com.yahoo.elide.datastores.aggregation.annotation.Meta;
+import com.yahoo.elide.datastores.aggregation.annotation.MetricAggregation;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.reflect.Constructor;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+
+/**
+ * {@link Metric} annotated by {@link MetricAggregation}.
+ * <p>
+ * {@link BaseMetric} is thread-safe and can be accessed by multiple threads.
+ */
+@Slf4j
+public class BaseMetric extends Column implements Metric {
+
+    /**
+     * Given a {@link Aggregation}, returns the SQL aggregation function.
+     * <p>
+     * For example, returns "MAX(%s)" if the specified {@link Aggregation} is {@link Max}. If the specified aggregation
+     * is {@code null}, this method returns an empty string, i.e. "".
+     *
+     * @param aggregationType  A sub-type of {@link Aggregation}, such as {@link Max} and {@link Min}.
+     *
+     * @return the SQL function format of the {@code aggregationType}
+     */
+    public static String functionFormat(Class<? extends Aggregation> aggregationType) {
+        if (aggregationType == null) {
+            return "";
+        }
+
+        try {
+            Class<?> clazz = Class.forName(aggregationType.getCanonicalName());
+            Constructor<?> ctor = clazz.getConstructor();
+            Aggregation instance = (Aggregation) ctor.newInstance();
+            return instance.getAggFunctionFormat();
+        } catch (Exception exception) {
+            String message = String.format(
+                    "Cannot generate aggregation function for '%s'",
+                    aggregationType.getCanonicalName()
+            );
+            log.error(message, exception);
+            throw new IllegalStateException(message, exception);
+        }
+    }
+
+    private static final long serialVersionUID = 3055820948480283917L;
+
+    @Getter
+    private final List<Class<? extends Aggregation>> aggregations;
+
+    @Getter
+    private final String metricExpression;
+
+    /**
+     * Constructor.
+     *
+     * @param metricField  The entity field or relation that this {@link Metric} represents
+     * @param annotation  Provides static meta data about this {@link Metric}
+     * @param fieldType  The Java type for this entity field or relation
+     * @param aggregations  A list of all supported aggregations on this {@link Metric}
+     * @param metricExpression  The SQL functions corresponding the default aggregation of this {@link Metric}. Note
+     * that the default aggregation is the first aggregation in this list
+     */
+    public BaseMetric(
+            String metricField,
+            Meta annotation,
+            Class<?> fieldType,
+            List<Class<? extends Aggregation>> aggregations,
+            String metricExpression
```

Comment:
I would remove this - and just compute at runtime.

Reply:
In fact, we might need to change the Metric.getMetricExpression interface to take an optional aggregation as an argument.